src/internal/connector/discovery/api/users.go

```diff
@@ -129,9 +144,18 @@ func (c Users) GetInfo(ctx context.Context, userID string) (*UserInfo, error) {
 
 	// TODO: OneDrive
 
-	_, err := c.stable.Client().UsersById(userID).MailFolders().Get(ctx, nil)
+	var (
+		err         error
+		internalErr error
+	)
+
+	err = graph.RunWithRetry(func() error {
+		_, internalErr = c.stable.Client().UsersById(userID).MailFolders().Get(ctx, nil)
+		return internalErr
+	})
+
 	if err != nil {
-		if !graph.IsErrExchangeMailFolderNotFound(err) {
+		if !graph.IsErrExchangeMailFolderNotFound(internalErr) {
```

Comment:
Same here with `internalErr`
```suggestion
		if !graph.IsErrExchangeMailFolderNotFound(err) {
```